self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryTestCase.java

```diff
@@ -0,0 +1,86 @@
+package com.selfxdsd.core.mock;
+
+import com.selfxdsd.api.Provider;
+import com.selfxdsd.api.Storage;
+import com.selfxdsd.api.User;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class InMemoryTestCase {
+
+    //users test cases
+    @Test
+    public void userSignUp() {
```

Comment:
@criske JavaDocs are needed on top of all tests. You can write simple javadocs with format: ``[ClassName] can [do something]". For Example:

```java
/*
 * InMemoryUsers can sign up one user.
 */
@Test
public void userSignUp() {
    //...
}
```